### PR TITLE
Jobs/Revocation prototype

### PR DESCRIPTION
**Pull request by johanbrandhorst**
This is a prototype implementation of https://docs.google.com/document/d/1tYBMlDJBdNA8POHyzkDWuyJjXUMV6SS23onBDn4Qg18

### CODE REVIEW

File: internal/db/schema/migrations/oss/postgres/42/02_add_job_config_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+begin;
+
+    create table job_config (
```
Comment (mgaffney) [inline internal/db/schema/migrations/oss/postgres/42/02_add_job_config_table.up.sql:3]: The table is missing a primary key

Comment (johanbrandhorst) [reply internal/db/schema/migrations/oss/postgres/42/02_add_job_config_table.up.sql:3]: Yeah I should probably just slap a `primary key (job_plugin_id, job_name)` to replace the `unique` below. Good note for the future implementation.

Comment (johanbrandhorst) [reply internal/db/schema/migrations/oss/postgres/42/02_add_job_config_table.up.sql:3]: Fixed

File: internal/db/schema/migrations/oss/postgres/42/01_create_kms_key_revocations.up.sql [hunk 2]
```diff
@@ -0,0 +1,43 @@
+begin;
+
+    create table kms_key_revocation_status_enm (
+        string text primary key
```
Comment (mgaffney) [inline internal/db/schema/migrations/oss/postgres/42/01_create_kms_key_revocations.up.sql:4]:
```suggestion
        name text primary key
```

File: internal/db/schema/migrations/oss/postgres/42/01_create_kms_key_revocations.up.sql [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        ('completed'),
+        ('failed');
+
+    create trigger 
+        mmutable_columns
+    before
+    update on kms_key_revocation_status_enm
+        for each row execute procedure immutable_columns('string');
+
+    create table kms_key_revocations(
```
Comment (mgaffney) [inline internal/db/schema/migrations/oss/postgres/42/01_create_kms_key_revocations.up.sql:25]:
```suggestion
    create table kms_key_revocation(
```
Table names should be singular

File: internal/kms/key_revocation.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	return [...]string{
+		"unspecified",
+		"pending",
+		"running",
+		"completed",
+		"failed",
+	}[k]
+}
+
+type KeyRevocation struct {
```
Comment (jimlambrt) [inline internal/kms/key_revocation.go:27]: it's a prototype... but I'm hopeful the implementation will have a factory for making these bits.

**Review by jimlambrt — COMMENTED** [review]
overall looks great.  I didn't do a super thorough review, since there's a coming non-prototype PR

**Comment by johanbrandhorst** [comment]
Closing pending redesign

**Pull request closed without merging**